showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseCodegenMetadataWriter.kt

```diff
@@ -63,16 +68,22 @@ internal class ShowkaseCodegenMetadataWriter(private val environment: XProcessin
         fileBuilder.build().writeTo(environment.filer, mode = XFiler.Mode.Aggregating)
     }
 
-    private fun createShowkaseCodegenMetadata(showkaseMetadata: ShowkaseMetadata) =
-        AnnotationSpec.builder(ShowkaseCodegenMetadata::class)
-            .addMember("showkaseName = %S", showkaseMetadata.showkaseName)
+    private fun createShowkaseCodegenMetadata(showkaseMetadata: ShowkaseMetadata): AnnotationSpec.Builder {
+        val  builder = AnnotationSpec.builder(ShowkaseCodegenMetadata::class)
             .addMember("showkaseGroup = %S", showkaseMetadata.showkaseGroup)
             .addMember("packageName = %S", showkaseMetadata.packageName)
             .addMember("packageSimpleName = %S", showkaseMetadata.packageSimpleName)
             .addMember("showkaseElementName = %S", showkaseMetadata.elementName)
             .addMember("insideObject = ${showkaseMetadata.insideObject}")
             .addMember("insideWrapperClass = ${showkaseMetadata.insideWrapperClass}")
             .addMember("showkaseKDoc = %S", showkaseMetadata.showkaseKDoc)
+        if (showkaseMetadata is ShowkaseMetadata.Component && showkaseMetadata.componentDistinctName != null) {
+            builder.addMember("showkaseName = %S", showkaseMetadata.componentDistinctName)
```

Comment:
won't this cause the name of the component to have index in the ShowkaseBrowser 🤔

Reply:
Yeah, this I'm somewhat unsure about. In the ShowkaseBrowser, I think it makes sense to distinguish them somehow. Here, I just used the `componentDistinctName`, but maybe it makes sense to use the preview name instead? 🤔 There is no check for the consumer if they would name the preview with the same names, but I think that would maybe be the best way to distinguish them? 🤔 Wdyt?

Reply:
Reverted this in [f185cec](https://github.com/airbnb/Showkase/pull/259/commits/f185cec407e6bc8103e3a378726d5b0eb01f215c) As there was no reason to have the index in the name in the ShowkaseBrowser 😄  Thanks for pointing this out!